packages/glimmer-runtime/lib/environment.ts

```diff
@@ -121,10 +122,19 @@ export abstract class Environment {
     return intern(ensureGuid(object) + '');
   }
 
+  expression(key: string, expression: ExpressionSyntax<Opaque>): ExpressionSyntax<Opaque> {
+    return this.refineExpression(parseExpression(key, expression)) || expression;
+  }
+
   statement(statement: StatementSyntax): StatementSyntax {
     return this.refineStatement(parseStatement(statement)) || statement;
   }
 
+  protected refineExpression(expression: ParsedExpression): ExpressionSyntax<Opaque> {
+    console.log(expression);
```

Comment:
You are passed something like 
<img width="522" alt="screen shot 2016-04-16 at 10 15 32 am" src="https://cloud.githubusercontent.com/assets/183799/14582529/727dc8fe-03bc-11e6-9744-2bb6ef710aa6.png">